airbyte-webapp/src/pages/ConnectionPage/pages/CreationFormPage/CreationFormPage.tsx

```diff
@@ -211,13 +211,16 @@ export const CreationFormPage: React.FC = () => {
           },
         ];
 
-  const titleId: string = (
-    {
-      [EntityStepsTypes.CONNECTION]: "connection.newConnectionTitle",
-      [EntityStepsTypes.DESTINATION]: "destinations.newDestinationTitle",
-      [EntityStepsTypes.SOURCE]: "sources.newSourceTitle",
-    } as Record<EntityStepsTypes, string>
-  )[type];
+  const titleId: string =
+    currentStep === "createConnection"
+      ? "connection.newConnectionTitle"
+      : (
+          {
+            [EntityStepsTypes.CONNECTION]: "connection.newConnectionTitle",
+            [EntityStepsTypes.DESTINATION]: "destinations.newDestinationTitle",
+            [EntityStepsTypes.SOURCE]: "sources.newSourceTitle",
+          } as Record<EntityStepsTypes, string>
+        )[type];
```

Comment:
This works but I hate how this is handled across this component.

Reply:
Agree.  I started looking at cleaning up some of the types and names here, but it snowballed and got pretty out of scope pretty quickly.